mongo/gridfs/bucket.go

```diff
@@ -455,6 +452,47 @@ func (b *Bucket) findChunks(ctx context.Context, fileID interface{}) (*mongo.Cur
 	return chunksCursor, nil
 }
 
+// returns true if the 2 index documents are equal
+func indexDocsEqual(expected, actual bsoncore.Document) (bool, error) {
+	if bytes.Equal(expected, actual) {
+		return true, nil
+	}
+
+	keyElemElems, err := actual.Elements()
```

Comment:
I think we can change the variable names from `keyElemElems` and `modelKeysElems` to `actualElems` and `expectedElems` for readability. That was the motivation for my suggestion to split this into a separate function.